flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -153,9 +165,90 @@ public ElasticsearchSink(Map<String, String> userConfig, List<InetSocketAddress>
 	 */
 	@Override
 	public void open(Configuration configuration) {
+		connect();
```

Comment:
I think we should try to avoid side-effecting class field members within utility member functions. It reduces the readability of the code. Return the created TransportClient instead, and have a separate utility function to check if connection is lost?